src/scriptworker/context.py

```diff
@@ -15,6 +15,7 @@
 import logging
 import os
 import tempfile
+import time
```

Comment:
at first glance I was going to suggest just using datetime and timedelta for better readability, but after a second pass on this, I'm ok with using numeric calculations with simpler code.